ee/clickhouse/queries/groups_join_query.py

```diff
@@ -29,10 +27,16 @@ def __init__(
         self._team_id = team_id
         self._column_optimizer = column_optimizer or ColumnOptimizer(self._filter, self._team_id)
 
-    def get_join_query(self) -> Tuple[str, Dict]:
+    def get_join_query(self, group_join_keys: Optional[List[str]] = None) -> Tuple[str, Dict]:
         join_queries, params = [], {}
 
-        for group_type_index in self._column_optimizer.group_types_to_query:
+        if group_join_keys:
```

Comment:
It's kind of iffy how the size of this needs to match `group_types_to_query` exactly - it creates a very tight coupling of things that shouldn't really be coupled (queries shouldn't know internals of column_optimizer).

Why is this needed? Can we decouple it somehow?

Looking at the callsite there'll only be a single argument passed so why not `str` only?

Reply:
My propsal is here: https://github.com/PostHog/posthog/pull/7056/files#r748075864